sqldb/sqlc/migrations/000001_invoices.up.sql

```diff
@@ -129,26 +148,8 @@ CREATE TABLE IF NOT EXISTS invoice_htlc_custom_records (
     value BLOB NOT NULL,
 
     -- The htlc id this record belongs to. 
-    htlc_id BIGINT NOT NULL REFERENCES invoice_htlcs(id)
+    htlc_id BIGINT NOT NULL REFERENCES invoice_htlcs(id) ON DELETE CASCADE
 );
 
 CREATE INDEX IF NOT EXISTS invoice_htlc_custom_records_htlc_id_idx ON invoice_htlc_custom_records(htlc_id);
 
--- invoice_payments contains the information of a settled invoice payment. 
-CREATE TABLE IF NOT EXISTS invoice_payments (
```

Comment:
This we removed to not have to do a join to figure out if an invoice has been settled?

Reply:
I _do_ think it's worth while to explore the event-sourced idea at this phase. The current rigidity of `settle_index` and `add_index` is what caused us to not try to also add events for stuff like accepting a hodl invoice. If we use an events table, then we can easily extend with other events in the future.